conn.go

```diff
@@ -387,11 +387,15 @@ func (c *boltConn) Close() error {
 		return nil
 	}
 
-	err := c.conn.Close()
-	c.closed = true
-	if err != nil {
-		c.connErr = errors.Wrap(err, "An error occurred closing the connection")
-		return driver.ErrBadConn
+	if c.conn != nil {
```

Comment:
This is a correct change. 

For more succinctness, I might suggest rewriting it as follows (since `c.closed = true` is on both if branches):

```go
c.closed = true
if c.conn != nil {
    if err := c.conn.Close(); err != nil {
        c.connErr = errors. ...
        return driver.ErrBadConn
    }
}
```